### PR TITLE
docs(breaking-change): Update user profile related docs after adding @loopback/security

### PR DESCRIPTION
**Pull request by jannyHou**
Fixes https://github.com/strongloop/loopback-next/issues/3655

## Checklist

👉 [Read and sign the CLA (Contributor License Agreement)](https://cla.strongloop.com/agreements/strongloop/loopback-next) 👈

- [ ] `npm test` passes on your machine
- [ ] New tests added or existing tests modified to cover all changes
- [ ] Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)
- [ ] API Documentation in code was updated
- [ ] Documentation in [/docs/site](../tree/master/docs/site) was updated
- [ ] Affected artifact templates in `packages/cli` were updated
- [ ] Affected example projects in `examples/*` were updated

👉 [Check out how to submit a PR](https://loopback.io/doc/en/lb4/submitting_a_pr.html) 👈

### CODE REVIEW

**Comment by derdeka** [comment]
`UserProfile` from `@loopback/security` requires a new field [securityId], therefor it will not be backwards compatible even with this export.

**Comment by jannyHou** [comment]
@derdeka True, I updated the docs as well, PTAL thanks.

**Comment by raymondfeng** [comment]
@jannyHou The newly released `@loopback/authentication` should have been a major as it contains breaking changes.

**Comment by jannyHou** [comment]
@raymondfeng 😞 True...how about we land this PR then release a new major version? Or should I revert the release(not sure is it even possible)

File: packages/authentication/src/index.ts [hunk 1]
```diff
@@ -3,6 +3,8 @@
 // This file is licensed under the MIT License.
 // License text available at https://opensource.org/licenses/MIT
 
+// Keep backward compatibility
+export {UserProfile} from '@loopback/security';
```
Comment (derdeka) [inline packages/authentication/src/index.ts]: @jannyHou you should remove the export if you want a semver major :-)

Comment (jannyHou) [reply packages/authentication/src/index.ts]: @derdeka reverted :)

**Comment by JohnPhoto** [comment]
Any info of how you will proceed, @jannyHou @raymondfeng?
This sort of broke my stuff and I am not sure if I should import the security-version and somehow patch the `securityId`  (any tips?) or downgrade, pin version and wait for a proper patch/minor version?

You can roll forward with a release of a patch with the revert to a non-breaking change and then deprecate the broken version.

**Comment by jannyHou** [comment]
@JohnPhoto Chatting with @raymondfeng , will fix it asap

**Comment by raymondfeng** [comment]
@derdeka Here is what we have done to unblock you:

1. Published `@loopback/authentication@2.2.2` with the same level code as `2.1.11`.
2. Deprecated `2.2.0` with a warning message
3. We'll publish `3.0.0` soon

File: docs/site/Loopback-component-authentication.md [hunk 2]
```diff
@@ -159,7 +155,7 @@ export class WhoAmIController {
   @authenticate('basic')
   @get('/whoami')
   whoAmI(): string {
-    return this.userProfile.id;
+    return this.userProfile[securityId];
```
Comment (raymondfeng) [inline docs/site/Loopback-component-authentication.md]: Let's add a comment saying `securityId` is `Symbol` to represent the security id of the user, typically the user id.

Comment (jannyHou) [reply docs/site/Loopback-component-authentication.md]: The link will be available when refresh the build of loopback.io

Comment (raymondfeng) [reply docs/site/Loopback-component-authentication.md]: Should the link be `https://loopback.io/doc/en/lb4/Security.html`?

Comment (jannyHou) [reply docs/site/Loopback-component-authentication.md]: @raymondfeng I think w/ or w/o `/html` both work, I removed it just to keep the link short.

Like https://loopback.io/doc/en/lb4/Components and https://loopback.io/doc/en/lb4/Components.html

**Pull request merged**